runtime/runtime.go

```diff
@@ -316,9 +319,26 @@ func NewRuntime(ctx context.Context, params Params) (*Runtime, error) {
 
 	metrics := prometheus.New(metrics.New(), errorLogger(logger))
 
+	var store storage.Store
+	if params.DiskStorage == nil {
+		params.DiskStorage, err = disk.OptionsFromConfig(config, params.ID)
+		if err != nil {
+			return nil, fmt.Errorf("parse disk store configuration: %w", err)
+		}
+	}
+
+	if params.DiskStorage != nil {
+		store, err = disk.New(ctx, *params.DiskStorage)
```

Comment:
💭 WDYT automatically creating the storage directory if it does not exist?

The immediate reason that comes to mind is that the docker image doesn't contain `mkdir` so users would need to volume mount a directory from the host into the container in order to use the disk storage... maybe this will be the typical deployment pattern, in which case it's a non-issue.

Having said that, I could imagine people wanting to use the disk-store without volume mounts just to reduce memory footprint, in which case auto-creating the directory would be nice...

Reply:
I think it's a decent idea! we'll check if the directory exists, and if it doesn't, we'll try to create it. if that fails, we error out. That should cover both cases.

The only slight annoyance I can foresee is typos in configs: OPA could create a new dir when it should really use an existing one, but had the path mis-typed. 

I guess the obvious option would be to add another configurable: auto-create, default to false. What do you think?

Reply:
✔️ Ended up not finding this comment until now, so it's become `auto_create` now. I can drop the underscore tomorrow.